armada/handlers/armada.py

```diff
@@ -109,96 +112,103 @@ def sync(self):
-                        post_actions = getattr(chart.upgrade, 'post', {})
-
-                # show delta for both the chart templates and the chart values
-                # TODO(alanmeadows) account for .files differences
-                # once we support those
-
-                upgrade_diff = self.show_diff(chart, installed_chart,
-                                              installed_values,
-                                              chartbuilder.dump(), values)
-
-                if not upgrade_diff:
-                    LOG.info("There are no updates found in this chart")
+            desc = entry.get('description', 'A Chart Group')
+            chart_group = entry.get('chart_group', [])
+
+            if entry.get('sequenced', False):
+                self.wait = True
+
+            LOG.info('Deploying: %s', desc)
+
+            for gchart in chart_group:
+                chart = dotify(gchart['chart'])
+                values = gchart.get('chart').get('values', {})
+                pre_actions = {}
+                post_actions = {}
+
+                if chart.release_name is None:
                     continue
 
-                # do actual update
-                self.tiller.update_release(protoc_chart,
-                                           self.dry_run,
-                                           chart.release_name,
-                                           chart.namespace,
-                                           prefix, pre_actions,
-                                           post_actions,
-                                           disable_hooks=chart.
-                                           upgrade.no_hooks,
-                                           values=yaml.safe_dump(values),
-                                           wait=self.wait,
-                                           timeout=chart_timeout)
-
-            # process install
-            else:
-                LOG.info("Installing release %s", chart.release_name)
-                self.tiller.install_release(protoc_chart,
-                                            self.dry_run,
-                                            chart.release_name,
-                                            chart.namespace,
-                                            prefix,
-                                            values=yaml.safe_dump(values),
-                                            wait=self.wait,
-                                            timeout=chart_timeout)
-
-            LOG.debug("Cleaning up chart source in %s",
-                      chartbuilder.source_directory)
-
-            chartbuilder.source_cleanup()
+                # retrieve appropriate timeout value if 'wait' is specified
+                chart_timeout = None
+                if self.wait:
+                    if self.timeout:
+                        chart_timeout = self.timeout
+                    elif getattr(chart, 'timeout', None):
```

Comment:
When would this path ever get activated -- a chart specific timeout? self.timeout has an oslo config default of 3600 so it would always be set?

Shouldn't this be:

```
if getattr(chart...)
  chart_timeout = chart.timeout
else
  chart_timeout = self.timeout
```